wallet/wallet.go

```diff
@@ -3706,6 +3713,83 @@ func (w *Wallet) LockedOutpoints() []dcrdtypes.TransactionInput {
 	return locked
 }
 
+func (w *Wallet) acctLockedOutpoints(ctx context.Context, acctName string) ([]dcrdtypes.TransactionInput, error) {
+	w.lockedOutpointMu.Lock()
+	locked := make([]wire.OutPoint, len(w.lockedOutpoints))
+	i := 0
+	for op := range w.lockedOutpoints {
+		locked[i] = op
+		i++
+	}
+	w.lockedOutpointMu.Unlock()
+
+	acctLocked := make([]dcrdtypes.TransactionInput, 0, len(locked))
+	err := walletdb.View(ctx, w.db, func(tx walletdb.ReadTx) error {
+		addrmgrNs := tx.ReadBucket(waddrmgrNamespaceKey)
+		txmgrNs := tx.ReadBucket(wtxmgrNamespaceKey)
+
+		defaultAccountName, err := w.manager.AccountName(
+			addrmgrNs, udb.DefaultAccountNum)
+		if err != nil {
+			return err
+		}
+
+		// Outpoints are not validated before they're locked, so
+		// invalid outpoints may be locked. Simply ignore.
+		for i := range locked {
+			op := &locked[i]
+			details, err := w.txStore.TxDetails(txmgrNs, &op.Hash)
+			if err != nil {
+				if errors.Is(err, errors.NotExist) {
+					// invalid txid
+					continue
+				}
+				return err
+			}
+			if int(op.Index) >= len(details.MsgTx.TxOut) {
+				// valid txid, invalid vout
+				continue
+			}
+			output := details.MsgTx.TxOut[op.Index]
+			// Lookup the associated account for the output.  Use the
+			// default account name in case there is no associated account
+			// for some reason, although this should never happen.
+			//
+			// This will be unnecessary once transactions and outputs are
+			// grouped under the associated account in the db.
+			opAcct := defaultAccountName
+			_, addrs, _, err := txscript.ExtractPkScriptAddrs(
+				0, output.PkScript, w.chainParams)
+			if err != nil {
+				continue
+			}
+			if len(addrs) > 0 {
+				acct, err := w.manager.AddrAccount(
+					addrmgrNs, addrs[0])
+				if err == nil {
+					s, err := w.manager.AccountName(
+						addrmgrNs, acct)
+					if err == nil {
+						opAcct = s
+					}
+				}
+			}
+			if opAcct == acctName {
+				acctLocked = append(acctLocked, dcrdtypes.TransactionInput{
+					Txid: op.Hash.String(),
+					Vout: op.Index,
+				})
```

Comment:
```suggestion
				acctLocked = append(acctLocked, dcrdtypes.TransactionInput{
					Amount: dcrutil.Amount(output.Value).ToCoin(),
					Txid:   op.Hash.String(),
					Vout:   op.Index,
					Tree:   op.Tree,
				})
```

Reply:
Ohh, actually `op.Tree` isn't even used when locking the output in `(*Server).lockUnspent`.  Looks like the tree will need to be inferred from `details.TxType`.